pkg/persistentcache/persistentcache.go

```diff
@@ -0,0 +1,61 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2019 Datadog, Inc.
+
+package persistentcache
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// Return a file where to store the data. We split the key by ":", using the
+// first prefix as directory, if present. This useful for integrations, which
+// use the check_id formed with $check_name:$hash
+func getFileForKey(key string) (string, error) {
```

Comment:
We need to sanitize `key` for sequences such as `/`, `../` to avoid potential security issues if we have an integration requiring part of the key to be controlled by something external.